From 1fc82ea63f16efc0d607b39a162e3032d7c2d602 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Wed, 3 Sep 2014 11:21:01 +0200
Subject: [PATCH] Use AC_PATH_TOOL to find libgcrypt-config and xml2-config

Using AC_PATH_TOOL prefers ${CHOST}-prefixed tools over 'bare' ones,
therefore improving support for cross-compilation and multilib. This
is useful for environment where libgcrypt-config and xml2-config
correspond to the build host environment, while
${CHOST}-libgcrypt-config and ${CHOST}-xml2-config are installed for
each supported build targets.

Fixes: https://bugzilla.gnome.org/show_bug.cgi?id=725635
Fixes: https://bugs.gentoo.org/show_bug.cgi?id=518728
---
 configure.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/configure.in b/configure.in
index a861e90..f1639a8 100644
--- a/configure.in
+++ b/configure.in
@@ -393,7 +393,7 @@ case $host in
     CRYPTO_TESTDIR=crypto
     ;;
   *)
-    AC_PATH_PROG(LIBGCRYPT_CONFIG, libgcrypt-config, no)
+    AC_PATH_TOOL(LIBGCRYPT_CONFIG, libgcrypt-config, no)
     if test "$LIBGCRYPT_CONFIG" != "no" ; then
       LIBGCRYPT_VERSION=`$LIBGCRYPT_CONFIG --version`
       if test VERSION_TO_NUMBER(echo $LIBGCRYPT_VERSION) -lt VERSION_TO_NUMBER(echo "1.1.42")
@@ -507,9 +507,9 @@ AC_SUBST(LIBXML_REQUIRED_VERSION)
 AC_MSG_CHECKING(for libxml libraries >= $LIBXML_REQUIRED_VERSION) 
 if test "x$LIBXML_CONFIG_PREFIX" != "x"
 then
-	XML_CONFIG=${LIBXML_CONFIG_PREFIX}/bin/xml2-config
+	AC_PATH_TOOL([XML_CONFIG], [xml2-config], [false], [${LIBXML_CONFIG_PREFIX}/bin])
 else
-	XML_CONFIG=xml2-config
+	AC_PATH_TOOL([XML_CONFIG], [xml2-config], [false])
 fi
 
 dnl
-- 
2.1.0

